include/numerics/petsc_vector.h

```diff
@@ -854,8 +951,6 @@ void PetscVector<T>::clear ()
     }
 
   this->_is_closed = this->_is_initialized = false;
-
-  _global_to_local_map.clear();
```

Comment:
I hadn't thought about NumericVector::clear() when I was blithely reassuring you that we wouldn't have any problems with sharing this map ... We definitely can't actually clear a shared map here, but if we leave the map untouched, then we aren't entirely cleared, in a sense.  Would it make sense to reset the map to nullptr?

Reply:
> Would it make sense to reset the map to nullptr?

Yes, or simply `_global_to_local_map.reset();` should do the right thing. If the internal reference count is reduced to zero by that call, then it will call the underlying object's destructor.